src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -260,11 +263,42 @@ public DatabaseTarget(string name) : this()
         [DefaultValue(CommandType.Text)]
         public CommandType CommandType { get; set; }
 
+        /// <summary>
+        /// Gets or sets property name of the SQL command parameter to set parameter DbType.
+        /// </summary>
+        /// <remarks>
+        /// May set strong DbType, for SQL Server is SqlDbType.
+        /// </remarks>
+        /// <docgen category='SQL Statement' order='12' />
+        [DefaultValue("DbType")]
+        public string ParameterDbTypePropertyName { get; set; }
+
+        /// <summary>
+        /// Gets or sets converter type of the SQL command parameter value.
+        /// </summary>
+        /// <docgen category='SQL Statement' order='13' />
+        [DefaultValue(typeof(DatabaseParameterTypeSetter))]
+        public Type ParameterConverterType { get; set; }
+
+        ///<summary>SQL Command Parameter Converter</summary>
+        public DatabaseParameterTypeSetter ParameterTypeSetter { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public IDatabaseValueConverter ParameterValueConverter
```

Comment:
@snakefoot do you think I have to move this to the LogFactory? I prefer injection without creating a subclass of the target.

Another option would be to add a `static` field here

Reply:
Think the DatabaseConverter is very DatabaseTarget specific, so not a direct static variable on LogFactory. Maybe consider extending the LogFactory to have a simple dependency-injection system? Instead of having these global-instances (Like IJsonConverter), then it could be neat to register a factory-method for a random interface-type-implementation.

Reply:
a static (with interface) is a simple dependency-injection system? ;)

Or should I inject it to the database target from ...?

Reply:
or are you proposing our own container and containerbuilder?

Reply:
Prefer static instance on the DatabaseTarget then

Reply:
We already have lots of containers. They are just hardcoded to Targets, Layouts, etc. Could easily have a container that could handle any interface-type.

Reply:
> Could easily have a container that could handle any interface-type-string.

Sure, but then we need a register and resolve (see https://github.com/NLog/NLog/pull/2899#discussion_r215438293).

And then i'm afraid we need (in the future), DI scopes, DI lifetimes, cached lookups, lambda support, constructor injection, etc etc.. (reinventing the wheel)

Reply:
You could also just support a DependencyInjection-interface, allowing people to come with their own dependency-injection-container.